kolibri/core/assets/src/state/modules/core/actions.js

```diff
@@ -555,6 +561,22 @@ export function updateTimeSpent(store, forceSave = false) {
   }
 }
 
+/**
+ * Update the content state in the summary log
+ * To be called periodically by interval timer
+ * Must be called after initContentSession
+ * @param {boolean} forceSave
+ */
+export function updateContentState(store, contentState, forceSave = false) {
```

Comment:
What are the use cases for using or not using force save?

Reply:
It's basically used as protection from overwhelming the database with a bunch of rapid changes to the content state, similar to the logic in updateProgress. (I basically cribbed that logic.) forceSave is to be used when you need an immediate update and the caller has been careful to ensure messages aren't coming from multiple points, which is why it defaults to false. 

Note that due to the latency of the HTML5 app <-> Kolibri message passing, Hashi itself has an app-local copy of the localStorage that exists during the app lifetime, so writes are not usually time-critical. The situation may be different for your ePub reader, and since you don't have to deal with things like postMessage, using forceSave may be a better choice there.